x/photon/keeper/msg_server.go

```diff
@@ -0,0 +1,113 @@
+package keeper
+
+import (
+	"context"
+
+	"cosmossdk.io/errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	govtypes "github.com/atomone-hub/atomone/x/gov/types"
+	"github.com/atomone-hub/atomone/x/photon/types"
+)
+
+type msgServer struct {
+	Keeper
+}
+
+// NewMsgServerImpl returns an implementation of the MsgServer interface
+// for the provided Keeper.
+func NewMsgServerImpl(keeper Keeper) types.MsgServer {
+	return &msgServer{Keeper: keeper}
+}
+
+var _ types.MsgServer = msgServer{}
+
+// MintPhoton implements the MsgServer.MintPhoton method.
+func (k msgServer) MintPhoton(goCtx context.Context, msg *types.MsgMintPhoton) (*types.MsgMintPhotonResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	params := k.GetParams(ctx)
+	if params.MintDisabled {
+		return nil, types.ErrMintDisabled
+	}
+
+	// Ensure burned amount denom is bond denom
+	bondDenom := k.stakingKeeper.BondDenom(ctx)
+	if msg.Amount.Denom != bondDenom {
+		return nil, types.ErrBurnInvalidDenom
+	}
```

Comment:
check also that there is a non-zero amount for the denom coin

Reply:
I would have still added the check here too but ok, I guess we'll check with a live test if this is a problem or not. I think you are right but let's see. Closing for now